src/engraving/rendering/score/lyricslayout.cpp

```diff
@@ -609,20 +609,23 @@ void LyricsLayout::setDefaultPositions(staff_idx_t staffIdx, const LyricsVersesM
     double staffHeight = ctx.dom().staff(staffIdx)->staffHeight();
     double lyricsLineHeightFactor = ctx.conf().styleD(Sid::lyricsLineHeight);
 
-    int totVersesAbove = int(lyricsVersesAbove.size()) - 1;
```

Comment:
As far as I can tell, the first commit can be stripped down to just this line. The error is assuming that the total number of verses coincides with the maximum verse (which is not always the case), so this should be
`int totVersesAbove = lyricsVersesAbove.empty() ? 0 : lyricsVersesAbove.rbegin()->first;`
as you rightly did. Perhaps better rename totVersesAbove as maxVerseAbove, which is more accurate.
The changes you've done to the loop below aren't changing anything, as far as I can tell, cause I'm running the loop forward with increasing y and you're running it backwards with decreasing y, which produces the same result. You could double check this by commenting out the function `checkCollisionWithStaffElements`, which lets you check the direct result of `setDefaultPositions`.

Reply:
Yes, you're correct! I'll remove the unecessary changes.